internal/resources/cluster/resource_cluster.go

```diff
@@ -594,6 +668,26 @@ func resourceClusterInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m
 
 		log.Printf("[INFO] cluster update successful")
 	}
+	// check default nodepool configuration update
+	npFullName := constructNpFullName(d)
```

Comment:
While adding the support for update , do we need to add support for delete as well ? I am not sure if deleting the default nodepool is supported when there are additional node pools available. So it depends on if API supports the deletion of default nodepool

Reply:
it does look like you can delete the default nodepool after another node pool is added. I'll look into how this might be done

Reply:
I think this would be a much bigger change since there is a check to ensure certain fields exist within the spec and the nodepool is one of them. this would require changing a lot more. I think we should table this for another update if it's needed.

Reply:
Ok @willisren - couple of questions:
1.  Does the API implementation takes care of throwing error when the there is only one nodpool in the cluster (say default nodepool created at the time of cluster create) and user tries to delete it?
2. If there are multiple node pools and the user tries to update the default(or any) nodepool with worker node count of 0, is this operation supported via nodepool Update API? - what exactly happens in this scenario?

Also @warroyo - I understand adding delete support might require more changes so, can you let us know how soon this is required from TMC terraform provider? Is there an urgent customer requirement on this?

Reply:
cc: @vmw-vjn

Reply:
1. Yes，there will be an error "the last node pool cannot be deleted" thrown out if users try to delete the last nodepool.
2. No, there will be an error "worker count is required to be greater than 0" thrown out if users try to update the replicas to 0 for a nodepool unless the autoscaler is enabled on it.